### PR TITLE
test: adds Alert action functional tests

### PR DESCRIPTION
**Pull request by carlossteinzup**
### Related Issues

- https://github.com/ZupIT/beagle/issues/1000
- https://github.com/ZupIT/beagle/issues/500

### Description and Example

This branch adds the functional tests to the Alert Action according to the Scenarios set for all fronts
- Android
- iOS
- WEB

### Checklist

Please, check if these important points are met using `[x]`:

- [x] I read the [PR Guide] and followed the process outlined there for submitting this PR.
- [x] I avoided _breaking changes_ by not changing public APIs that people rely on. 
- [x] I am willing to follow-up on review comments in a timely manner.

[PR Guide]: https://github.com/ZupIT/beagle/blob/master/doc/contributing/pull_requests.md

### CODE REVIEW

**Comment by botbeaglezup** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Big PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: iOS/Tests/AutomatedTestsUITests/Steps/AlertSteps.swift [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import Foundation
+import XCTest
+
+class AlertSteps: CucumberStepsDefinition {
+    var application: XCUIApplication!
```
Comment (botbeaglezup) [inline iOS/Tests/AutomatedTestsUITests/Steps/AlertSteps.swift:22]: <table data-meta="generated_by_ios">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/48a055f136986ff5f370d429870228e1ba4fe60c/iOS/Tests/AutomatedTestsUITests/Steps/AlertSteps.swift#L22"></span>Implicitly unwrapped optionals should be avoided when possible.<br />
<code>implicitly_unwrapped_optional</code> <code>AlertSteps.swift:22</code></td>
    </tr>
  </tbody>
</table>

File: iOS/Tests/AutomatedTestsUITests/Steps/AlertSteps.swift [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        // MARK: - Given
+        Given("^the Beagle application did launch with the alert screen url$") { _, _ -> Void in
+            XCTAssertTrue(ScreenElements.ALERT_SCREEN_TITLE.element.exists)
+        }
+        
+        // MARK: - When
+        
+        // Scenarios 1, 2, 3 and 4
+        When(#"^I press an alert button with the "([^\"]*)" title$"#) { args, _ -> Void in
+            let text = args![0]
```
Comment (botbeaglezup) [inline iOS/Tests/AutomatedTestsUITests/Steps/AlertSteps.swift:42]: <table data-meta="generated_by_ios">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/48a055f136986ff5f370d429870228e1ba4fe60c/iOS/Tests/AutomatedTestsUITests/Steps/AlertSteps.swift#L42"></span>Force unwrapping should be avoided.<br />
<code>force_unwrapping</code> <code>AlertSteps.swift:42</code></td>
    </tr>
  </tbody>
</table>

File: iOS/Tests/AutomatedTestsUITests/Steps/AlertSteps.swift [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+            self.application.buttons[text].tap()
+        }
+        
+        // MARK: - Then
+        
+        // Scenario 1
+        Then(#"^an alert with the "([^\"]*)" message should appear on the screen$"#) { args, _ -> Void in
+            let text = args![0]
```
Comment (botbeaglezup) [inline iOS/Tests/AutomatedTestsUITests/Steps/AlertSteps.swift:50]: <table data-meta="generated_by_ios">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/48a055f136986ff5f370d429870228e1ba4fe60c/iOS/Tests/AutomatedTestsUITests/Steps/AlertSteps.swift#L50"></span>Force unwrapping should be avoided.<br />
<code>force_unwrapping</code> <code>AlertSteps.swift:50</code></td>
    </tr>
  </tbody>
</table>

File: iOS/Tests/AutomatedTestsUITests/Steps/AlertSteps.swift [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+            XCTAssertTrue(self.application.staticTexts[text].exists)
+        }
+        
+        // Scenario 2
+        Then(#"^an alert with the "([^\"]*)" title and the "([^\"]*)" message should appear on the screen$"#) { args, _ -> Void in
+            let title = args![0]
```
Comment (botbeaglezup) [inline iOS/Tests/AutomatedTestsUITests/Steps/AlertSteps.swift:56]: <table data-meta="generated_by_ios">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/48a055f136986ff5f370d429870228e1ba4fe60c/iOS/Tests/AutomatedTestsUITests/Steps/AlertSteps.swift#L56"></span>Force unwrapping should be avoided.<br />
<code>force_unwrapping</code> <code>AlertSteps.swift:56</code></td>
    </tr>
  </tbody>
</table>

File: iOS/Tests/AutomatedTestsUITests/Steps/AlertSteps.swift [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+            let message = args![1]
```
Comment (botbeaglezup) [inline iOS/Tests/AutomatedTestsUITests/Steps/AlertSteps.swift:57]: <table data-meta="generated_by_ios">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/48a055f136986ff5f370d429870228e1ba4fe60c/iOS/Tests/AutomatedTestsUITests/Steps/AlertSteps.swift#L57"></span>Force unwrapping should be avoided.<br />
<code>force_unwrapping</code> <code>AlertSteps.swift:57</code></td>
    </tr>
  </tbody>
</table>

File: iOS/Tests/AutomatedTestsUITests/Steps/AlertSteps.swift [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+            XCTAssertTrue(self.application.staticTexts[title].exists)
+            XCTAssertTrue(self.application.staticTexts[message].exists)
+        }
+        
+        // Scenario 3
+        Then(#"^a second alert with the "([^\"]*)" title should appear on the screen$"#) { args, _ -> Void in
+            let text = args![0]
```
Comment (botbeaglezup) [inline iOS/Tests/AutomatedTestsUITests/Steps/AlertSteps.swift:64]: <table data-meta="generated_by_ios">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/48a055f136986ff5f370d429870228e1ba4fe60c/iOS/Tests/AutomatedTestsUITests/Steps/AlertSteps.swift#L64"></span>Force unwrapping should be avoided.<br />
<code>force_unwrapping</code> <code>AlertSteps.swift:64</code></td>
    </tr>
  </tbody>
</table>

File: iOS/Tests/AutomatedTestsUITests/Steps/AlertSteps.swift [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+            let element = self.application.staticTexts[text]
+            let result = element.waitForExistence(timeout: 10)
+            XCTAssertNotNil(result)
+            XCTAssertTrue(element.exists)
+        }
+        
+        // Scenario 4
+        Then(#"^an alert should appear on the screen and its confirmation button should display the "([^\"]*)" custom title$"#) { args, _ -> Void in
+            let title = args![0]
```
Comment (botbeaglezup) [inline iOS/Tests/AutomatedTestsUITests/Steps/AlertSteps.swift:73]: <table data-meta="generated_by_ios">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/48a055f136986ff5f370d429870228e1ba4fe60c/iOS/Tests/AutomatedTestsUITests/Steps/AlertSteps.swift#L73"></span>Force unwrapping should be avoided.<br />
<code>force_unwrapping</code> <code>AlertSteps.swift:73</code></td>
    </tr>
  </tbody>
</table>

File: iOS/Tests/AutomatedTestsUITests/Steps/AlertSteps.swift [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+            let element = self.application.buttons[title]
+            XCTAssertTrue(element.exists)
+        }
+    }
+}
+
```
Comment (botbeaglezup) [inline iOS/Tests/AutomatedTestsUITests/Steps/AlertSteps.swift:79]: <table data-meta="generated_by_ios">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/48a055f136986ff5f370d429870228e1ba4fe60c/iOS/Tests/AutomatedTestsUITests/Steps/AlertSteps.swift#L79"></span>Files should have a single trailing newline.<br />
<code>trailing_newline</code> <code>AlertSteps.swift:79</code></td>
    </tr>
  </tbody>
</table>

**Review by gabcoelho — APPROVED** [review]

**Review by matheusribeirozup — APPROVED** [review]

**Review by lucasfonsecazup — APPROVED** [review]

**Pull request merged**